src/textual/renderables/text_opacity.py

```diff
@@ -67,11 +68,16 @@ def process_segments(
 
             color = style.color
             bgcolor = style.bgcolor
-            if color and color.triplet and bgcolor and bgcolor.triplet:
-                color_style = _get_blended_style_cached(bgcolor, color, opacity)
-                yield _Segment(text, style + color_style)
+
+            if opacity > 0:
```

Comment:
This condition will be constant in the loop. If we move it outside of the loop then we can write an optimized version for the case of opacity == 0